File: examples/typescript/events/list-event-destination-events.ts
/**
 * List the most recent events in the account.
 *
 *   STEDI_API_KEY=<api-key> tsx list-event-destination-events.ts
 *
 * Reads a single page; creates and deletes nothing.
 */

import { Stedi } from "@stedi/sdk";

const PAGE_SIZE = 5;

const apiKey = process.env["STEDI_API_KEY"];
if (!apiKey) {
  throw new Error("STEDI_API_KEY is not set");
}

const client = new Stedi({ apiKey: { apiKey } });

const page = await client.listEventDestinationEvents({ pageSize: PAGE_SIZE });
const events = page.items ?? [];

console.log(`${events.length} of the most recent event(s):`);
for (const event of events) {
  console.log(`  ${event.id} [${event.status}] ${event.eventType}`);
  console.log(`    created ${event.createdAt?.toISOString()}`);
}
console.log(page.nextPageToken ? "more pages available" : "no further pages");

File: examples/python/events/list_event_destination_events.py
# /// script
# requires-python = ">=3.12"
# dependencies = ["stedi==1.0.0"]
# ///

"""List the most recent events in the account.

    STEDI_API_KEY=<api-key> uv run --script list_event_destination_events.py

Reads a single page; creates and deletes nothing.
"""

import asyncio
import os

from stedi import Config, Stedi
from stedi.models import ListEventDestinationEventsInput

PAGE_SIZE = 5


async def main(api_key: str) -> None:
    async with Stedi(Config(api_key=api_key)) as client:
        page = await client.list_event_destination_events(
            ListEventDestinationEventsInput(page_size=PAGE_SIZE),
        )

    events = page.items or []
    print(f"{len(events)} of the most recent event(s):")
    for event in events:
        print(f"  {event.id} [{event.status}] {event.event_type}")
        print(f"    created {event.created_at:%Y-%m-%d %H:%M:%S}")
    print("more pages available" if page.next_page_token else "no further pages")


api_key = os.environ.get("STEDI_API_KEY")
if not api_key:
    raise SystemExit("STEDI_API_KEY is not set")

asyncio.run(main(api_key))
